lib/compress.js

```diff
@@ -803,6 +805,22 @@ merge(Compressor.prototype, {
                 }
             }
 
+            function extract_args() {
+                var iife, fn = compressor.self();
+                if (fn instanceof AST_Function
```

Comment:
That'll be `is_func_expr(fn)` on the harmony branch presumably?

Don't we also need a `&& !has_pure_annotation()` condition here?